source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -25,13 +25,20 @@ void ProcessorState::onStartProcessorCall(Event::TimerCb cb, std::chrono::millis
     message_timer_ = filter_callbacks_->dispatcher().createTimer(cb);
   }
   message_timer_->enableTimer(timeout);
+  call_start_time_ = filter_callbacks_->dispatcher().timeSource().monotonicTime();
 }
 
-void ProcessorState::onFinishProcessorCall(CallbackState next_state) {
-  callback_state_ = next_state;
+void ProcessorState::onFinishProcessorCall(absl::Status call_status, CallbackState next_state) {
   if (message_timer_) {
     message_timer_->disableTimer();
   }
+  if (call_start_time_.has_value()) {
```

Comment:
We expect call_start_time.has_value() to always be true when onFinishProcessorCall() is called, right? If so, it might be worth an ENVOY_BUG if this is ever not the case.

Hmm, though reading further, I wonder if there might be cases where onFinishProcessorCall() is invoked at multiple levels for a given stream. If that's the case, never mind the earlier comment.

Reply:
Thanks, SG. (I'd resolve this comment thread except that I can't seem to find the button to do so.)